src/autoscaler/eventgenerator/aggregator/metricPoller.go

```diff
@@ -104,17 +102,17 @@ func (m *MetricPoller) doAggregate(appId string, metricType string, metrics []*m
 		avgAppMetric = &model.AppMetric{
 			AppId:      appId,
 			MetricType: metricType,
-			Value:      0,
+			Value:      -1,
 			Unit:       "",
-			Timestamp:  0,
+			Timestamp:  time.Now().UnixNano(),
```

Comment:
My point is that you can define the timestamp before the loop and use a single common value for all generated metrics.